geopandas/plotting.py

```diff
@@ -36,12 +41,15 @@ def _flatten_multi_geoms(geoms, colors):
         else:
             components.append(geom)
             component_colors.append(color)
+
+    if was_none:
+        component_colors = None
```

Comment:
I like your suggestion for moving the logic (that's indeed a better separation of concerns).
However, I am not sure `if None in values` is completely safe, as you can in principle have None as a valid value? (although maybe matplotlib will raise then, have to check)